controllers/runnerpool_controller_test.go

```diff
@@ -441,4 +481,93 @@ var _ = Describe("RunnerPool reconciler", func() {
 		By("deleting the created RunnerPool")
 		deleteRunnerPool(ctx, runnerPoolName, namespace)
 	})
+
+	It("should or not update secret", func() {
+		By("deploying RunnerPool resource")
+		rp := makeRunnerPool(runnerPoolName, namespace, repositoryNames[0])
+		Expect(k8sClient.Create(ctx, rp)).To(Succeed())
+
+		By("wating the RunnerPool become Bound")
+		Eventually(func() error {
+			rp := new(meowsv1alpha1.RunnerPool)
+			if err := k8sClient.Get(ctx, types.NamespacedName{Name: runnerPoolName, Namespace: namespace}, rp); err != nil {
+				return err
+			}
+			if !rp.Status.Bound {
+				return errors.New(`status "bound" should be true`)
+			}
+			return nil
+		}).Should(Succeed())
+		time.Sleep(wait) // Wait for the reconciliation to run a few times. Please check the controller's log.
+
+		testCase := []struct {
+			expiresInDuration time.Duration
+			shouldUpdate      bool
+		}{
+			{
+				-10 * time.Minute,
+				true,
+			},
+			{
+				0 * time.Minute,
+				true,
+			},
+			{
+				5 * time.Minute,
+				true,
+			},
+			{
+				20 * time.Minute,
+				false,
+			},
+			{
+				1 * time.Hour,
+				false,
+			},
+		}
+		for _, tc := range testCase {
+			By("getting the created Secret")
+			fmt.Printf("testcase is {expiresInDuration: %s, should update: %v}\n", tc.expiresInDuration.String(), tc.shouldUpdate)
+			s := new(corev1.Secret)
+			Expect(k8sClient.Get(ctx, types.NamespacedName{Name: rp.GetRunnerSecretName(), Namespace: namespace}, s)).To(Succeed())
+
+			By("set annotation")
+			baseTime := time.Now().Add(tc.expiresInDuration).Format(time.RFC3339)
+			s.Annotations[constants.RunnerSecretExpiresInAnnotationKey] = baseTime
+			Expect(k8sClient.Update(ctx, s)).To(Succeed())
+			time.Sleep(wait)
+
+			if !tc.shouldUpdate {
+				By("checking to not update secret")
+				s := new(corev1.Secret)
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: rp.GetRunnerSecretName(), Namespace: namespace}, s)
+				Expect(err).ToNot(HaveOccurred())
+				tm := s.Annotations[constants.RunnerSecretExpiresInAnnotationKey]
+				Expect(err).ToNot(HaveOccurred())
+				Expect(tm).Should(Equal(baseTime))
+				continue
+			}
+			By("checking to update secret")
+			Eventually(func() error {
```

Comment:
Why do you use `Eventually`?
After updating the secret, you set a wait (L538). So the secret has been updated at this point.